src/TriangleNodesProvisioner.cpp

```diff
@@ -288,4 +291,157 @@ namespace blitzdg {
+                ++count;
+            }
+        }
+
+        // TODO: Move Fmask computation to a helper method.
+        index_matrix_type Fm = *Fmask.get();
+        Fm = 0*jj;
+        Fm(Range::all(), 0) = fmask1;
+        Fm(Range::all(), 1) = fmask2;
+        Fm(Range::all(), 2) = fmask3;
+    }
+
+    void TriangleNodesProvisioner::buildLift() {
+        firstIndex ii;
+        secondIndex jj;
+        thirdIndex kk;
+
+        real_matrix_type E(NumLocalPoints, NumFaces*NumFacePoints), MassInv(NumLocalPoints, NumLocalPoints);
+
+        real_matrix_type & Liftref = *Lift.get();
+
+        real_vector_type& r = *rGrid.get();
+        real_vector_type& s = *sGrid.get();
+
+        index_matrix_type Fm = *Fmask.get();
+
+
+        real_vector_type faceR(NumFacePoints), faceS(NumFacePoints);
+        real_matrix_type V1D(NumFacePoints, NumFacePoints), V1Dinv(NumFacePoints,NumFacePoints);
+        real_matrix_type massEdgeInv(NumFacePoints, NumFacePoints);
+
+        real_matrix_type massEdge1(NumFacePoints, NumFacePoints), massEdge2(NumFacePoints,NumFacePoints),
+                         massEdge3(NumFacePoints, NumFacePoints);
+
+        // Face 1
+        for (index_type i=0; i < NumFacePoints; ++i)
+            faceR(i) = r(Fm(i, 0));
+
+        std::cout << faceR << std::endl;
+
+        Vandermonde.computeVandermondeMatrix(faceR, V1D, V1Dinv);
+        massEdgeInv = sum(V1D(ii,kk)*V1D(jj,kk), kk);
+
+        std::cout << massEdgeInv << std::endl;
+
+        Inverter.computeInverse(massEdgeInv, massEdge1);
+
+        std::cout << massEdge1 << std::endl;
+
+        E = 0.0*jj;
+        for (index_type i=0; i < NumFacePoints; ++i) {
+            for (index_type j=0; j < NumFacePoints; ++j) {
+                E(Fm(i,0),j) = massEdge1(i,j);
+            }
+        }
+
+        // Face 2
+        for (index_type i=0; i < NumFacePoints; ++i)
+            faceR(i) = r(Fm(i, 1));
+        
+        std::cout << Fm(Range::all(), 1) << std::endl;
+        std::cout << faceR << std::endl;
+
+
+        Vandermonde.computeVandermondeMatrix(faceR, V1D, V1Dinv);
+        massEdgeInv = sum(V1D(ii,kk)*V1D(jj,kk), kk);
+
+        Inverter.computeInverse(massEdgeInv, massEdge2);
+        for (index_type i=0; i < NumFacePoints; ++i) {
+            for (index_type j=NumFacePoints; j < 2*NumFacePoints; ++j) {
+                E(Fm(i,1),j) = massEdge2(i,j % NumFacePoints);
+            }
+        }
+
+        // Face 3.
+        for (index_type i=0; i < NumFacePoints; ++i)
+            faceS(i) = s(Fm(i, 2));
+        
+
+        std::cout << faceS << std::endl;
+
+        Vandermonde.computeVandermondeMatrix(faceS, V1D, V1Dinv);
+        massEdgeInv = sum(V1D(ii,kk)*V1D(jj,kk), kk);
+
+        Inverter.computeInverse(massEdgeInv, massEdge3);
+        for (index_type i=0; i < NumFacePoints; ++i) {
+            for (index_type j=2*NumFacePoints; j < 3*NumFacePoints; ++j) {
+                E(Fm(i,2),j) = massEdge3(i,j % NumFacePoints);
```

Comment:
Replace `j % NumFacePoints` with `j - 2 * NumFacePoints`, similar to the comment above.